core/src/main/java/com/microsoft/applicationinsights/TelemetryClient.java

```diff
@@ -507,6 +497,7 @@ public void flush() {
     TelemetryChannel getChannel() {
         if (this.channel == null) {
             this.channel = configuration.getChannel();
+            SDKShutdownActivity.INSTANCE.register(this.channel);
```

Comment:
I think this should still be in the constructor, `getChannel()` since it's called in `track()`.